CMakeLists.txt

```diff
@@ -95,8 +95,49 @@ ELSE()
   SET(CMAKE_THREAD_PREFER_PTHREAD ON)
   FIND_PACKAGE(Threads REQUIRED)
   ADD_COMPILE_FLAG("-std=c++11")
-  ADD_COMPILE_FLAG("-msse2")
-  ADD_COMPILE_FLAG("-mfpmath=sse")
+  if (NOT EMSCRIPTEN)
+    # try to get the target architecture by compiling a dummy.c file and
+    # checking the architecture using the file command.
+    file(WRITE ${PROJECT_BINARY_DIR}/dummy.c "main(){}")
```

Comment:
maybe worth logging something out here and later down? then if it fails on some platform, that might help figure things out. We already have a bunch of logs (`Building with` etc.) anyhow.